### ISSUE TITLE
embed code breaks when inserted on page which loads api script or jquery < 1.7.1

### ISSUE THREAD

**Issue by foliovision** [foliovision/freedom-video-player#212]
Original issue link: [#243 embed code breaks when inserted on page which loads api script or jquery < 1.7.1](https://github.com/flowplayer/flowplayer/issues/243)

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Feb 3, 2013

Also affects page layout
See: http://example.flowplayer.org/bug/243-1.html
Probably the embed script must bail out if `typeof flowplayer === "function"`.
See also: https://github.com/flowplayer/work/issues/607

EDIT: updated bug title

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Feb 8, 2013

Embedding fails without error when a page loads jquery < 1.7.1 - not good.
Looks like in the long run we MUST make flowplayer.js independent of jquery.
You can't control or give technical advice regarding requirements where embedding works.
This kind of problem is _extremely_ annoying for content offerers. They will discover it only after some time, and when it's too late. It's like something bad happening behind their backs.
Elevating to critical.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/1395672?v=4' /> [bbbo](https://github.com/bbbo) commented on Feb 8, 2013

and we must move the embed script from flowplayer.org into the 
flowplayer distribution, at least as an alternative.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Feb 8, 2013

See: http://flowplayer.org/forum/#!/moot/5418

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Feb 13, 2013

Embed docs claim:

---

The script loads following files: jQuery, flowplayer.js, Google Analytics and flowplayer skin only if they are not present on the embedding page. [...]

---

Looks like this claim is wrong.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Feb 13, 2013

No, it seems also on our page it's some kind of jquery conflict - even when I do not load jquery on the page. Embedding works once I load jquery 1.8(!) - although the last version which had trouble with 1.9 was 5.2.0.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/1395672?v=4' /> [bbbo](https://github.com/bbbo) commented on Feb 17, 2013

we definitely need a separate embed plugin which does not use embed.flowplayer.org (bad idea anyway for  multiple reasons); this will automatically get rid of such conflicts. 

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Feb 17, 2013

@bbbo - if a page already loads jquery < 1.7.1 (CMS) this won't help much either. But yes.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/1395672?v=4' /> [bbbo](https://github.com/bbbo) commented on Feb 17, 2013

made a thinking error; of course the embed script would need to default to some jQuery version to load for the embedded player. and it cannot know different versions present on the target page :(

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Feb 17, 2013

2 sample pages:
http://example.flowplayer.org/bug/243-1.html  - api script already loaded on page
http://example.flowplayer.org/bug/243-2.html - incompatible jquery version loaded on page

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/3911470?v=4' /> [JoMas971](https://github.com/JoMas971) commented on Apr 18, 2013

Hi,
I've noted also, that when you try to use the code embed on a CMS (Wordpress or Drupal), if you doesn't have any jQuery librairie loaded, the embed code doesn't work at all ! Nothing appear on my post. 
Maybe using an iframe is the solution or implement #291. Have you planed a release to fix this bug soon ?
Tanks.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Apr 18, 2013

@JoMas971 - the embed code does load jquery, but in a cms there might be conflicts which cause the player not to appear, which is what this bug is about.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Apr 18, 2013

@JoMas971 - Here's the demo without the page loading jquery on its own: http://demos.flowplayer.org/basics/embedded.html

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/265438?v=4' /> [nnarhinen](https://github.com/nnarhinen) commented on Apr 22, 2013

@blacktrash Embedding doesn't actually break when loaded to a site with flowplayer.js available. It just doesn't load any skin then anymore. Which is IMO ok. Why would you have flowplayer.js without any sane skin?

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/265438?v=4' /> [nnarhinen](https://github.com/nnarhinen) commented on Apr 22, 2013

Please try with

```
<script src="http://embed.dev94.flowplayer.us/dev/embed.min.js"><div class="flowplayer" style="width: 624px; height: 260px;" data-origin="http://demos.dev94.flowplayer.us/basics/minimal.html"><video><source type="video/webm" src="http://stream.flowplayer.org/bauhaus/624x260.webm"><source type="video/mp4" src="http://stream.flowplayer.org/bauhaus/624x260.mp4"><source type="video/ogg" src="http://stream.flowplayer.org/bauhaus/624x260.ogv"></video></div></script>
```

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/265438?v=4' /> [nnarhinen](https://github.com/nnarhinen) commented on Apr 22, 2013

(also clear cache)

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Apr 22, 2013

@nnarhinen - same result.
See also our own test page: http://flowplayer.org/test/logo.html - embedded does not appear at all.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/265438?v=4' /> [nnarhinen](https://github.com/nnarhinen) commented on Apr 22, 2013

@blacktrash Did you actually try with my code using http://embed.dev94.flowplayer.us/dev/embed.min.js ?

Please open this script directly in your browser and refresh. 

http://flowplayer.org/test/logo.html atleast doesn't use that script.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Apr 22, 2013

@nnarhinen - yes, I copied

```
<script src="http://embed.dev94.flowplayer.us/dev/embed.min.js"><div class="flowplayer" style="width: 624px; height: 260px;" data-origin="http://demos.dev94.flowplayer.us/basics/minimal.html"><video><source type="video/webm" src="http://stream.flowplayer.org/bauhaus/624x260.webm"><source type="video/mp4" src="http://stream.flowplayer.org/bauhaus/624x260.mp4"><source type="video/ogg" src="http://stream.flowplayer.org/bauhaus/624x260.ogv"></video></div></script>
```

verbatim into a page which loads flowplayer.min.js and refreshed etc.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/265438?v=4' /> [nnarhinen](https://github.com/nnarhinen) commented on Apr 22, 2013

Yeah as I said there's no bug with embedding to a page with flowplayer-library without skin. It's just the player without skin. Nothing really there for me to do.

What I fixed, is embedding to a page where jQuery is loaded but is too old.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Apr 22, 2013

But isn't it still a bug when it's bound to fail on our own pages? iow http://demos.flowplayer.org/bug/243-1.html is a wontfix?

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/265438?v=4' /> [nnarhinen](https://github.com/nnarhinen) commented on Apr 22, 2013

Different problem. Already fixed in dev, most likely.

2013/4/22 Christian Ebert notifications@github.com

> But isn't it still a bug when it's bound to fail on our own pages? iow
> http://demos.flowplayer.org/bug/243-1.html is a wontfix?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/flowplayer/flowplayer/issues/243#issuecomment-16778282
> .

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/265438?v=4' /> [nnarhinen](https://github.com/nnarhinen) commented on Apr 22, 2013

There is nothing for me to fix in
http://demos.flowplayer.org/bug/243-1.html

It is such a corner case where someone would load flowplayer.(min.)js but
not a skin to his site that there is no idea doing _ugly_ half-functional
hacks for it. Or is there a valid use case for such actions?

2013/4/22 Niklas Närhinen niklas@narhinen.net

> Different problem. Already fixed in dev, most likely.
> 
> 2013/4/22 Christian Ebert notifications@github.com
> 
> > But isn't it still a bug when it's bound to fail on our own pages? iow
> > http://demos.flowplayer.org/bug/243-1.html is a wontfix?
> > 
> > —
> > Reply to this email directly or view it on GitHubhttps://github.com/flowplayer/flowplayer/issues/243#issuecomment-16778282
> > .

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Apr 22, 2013

One corner case is e.g. flowplayer.org ;-) - i.e. a site which loads scripts via a build system which are needed most of time on all pages.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/265438?v=4' /> [nnarhinen](https://github.com/nnarhinen) commented on Apr 22, 2013

the case of flowplayer.org/test/logo.html is completely different from
this. It's no css problem but a javascript error, most likely already fixed
in dev. Have to build dev-site to ensure.

Even if the css would be built, it _would_be_there_ and the player would be
fully functional.

2013/4/22 Christian Ebert notifications@github.com

> One corner case is e.g. flowplayer.org ;-) - i.e. a site which loads
> scripts via a build system which are needed most of time on all pages.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/flowplayer/flowplayer/issues/243#issuecomment-16779051
> .

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Apr 22, 2013

Confirming that http://demos.flowplayer.org/bug/243-2.html is solved.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/265438?v=4' /> [nnarhinen](https://github.com/nnarhinen) commented on Apr 22, 2013

http://dev94.flowplayer.us/test/logo.html

2013/4/22 Christian Ebert notifications@github.com

> Confirming that http://demos.flowplayer.org/bug/243-2.html is solved.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/flowplayer/flowplayer/issues/243#issuecomment-16779169
> .

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Apr 22, 2013

@nnarhinen - yes, className minimalist is manually added there - which might do the trick - or not - the same embed code in the 241-1.html demo - loading http://cdn.dev94.flowplayer.us/dev/commercial/flowplayer.min.js - still does not show the skin.
Feel free to close it.

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/265438?v=4' /> [nnarhinen](https://github.com/nnarhinen) commented on Apr 22, 2013

http://demos.flowplayer.org/bug/243-1.html doesn't load any skin (in
<head>). And embed.min.js cannot know it isn't there. It has to assume that
if flowplayer-library is present, then there must be a skin also.

2013/4/22 Christian Ebert notifications@github.com

> @nnarhinen https://github.com/nnarhinen - yes, className minimalist is
> manually added there - which might do the trick - or not - the same embed
> code in the 241-1.html demo - loading
> http://cdn.dev94.flowplayer.us/dev/commercial/flowplayer.min.js - still
> does not show the skin.
> Feel free to close it.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/flowplayer/flowplayer/issues/243#issuecomment-16780121
> .

----

<img width='32' height='32' src='https://avatars.githubusercontent.com/u/168338?v=4' /> [phloxic](https://github.com/phloxic) commented on Apr 22, 2013

Ah, sorry for being slow on the uptake. Yes, that's a reasonable assumption that a skin is loaded too on a page where flowplayer.js is loaded.
Closing - with thanks for your patience.
